core/src/main/java/org/elasticsearch/index/query/ConstantScoreQueryBuilder.java

```diff
@@ -66,20 +69,81 @@ public ConstantScoreQueryBuilder boost(float boost) {
         return this;
     }
 
+    /**
+     * @return the boost factor
+     */
+    public float boost() {
+        return this.boost;
+    }
+
     @Override
     protected void doXContent(XContentBuilder builder, Params params) throws IOException {
         builder.startObject(NAME);
         builder.field("filter");
-        filterBuilder.toXContent(builder, params);
+        queryBuilder.toXContent(builder, params);
+        builder.field("boost", boost);
+        builder.endObject();
+    }
 
-        if (boost != -1) {
-            builder.field("boost", boost);
+    @Override
+    public Query toQuery(QueryParseContext parseContext) throws QueryParsingException, IOException {
+        if (queryBuilder == null ) {
+            return null;
```

Comment:
That's the way it was in the former parser() code in the parser. There `null` was returned when the inner filter parses to `null`. I only added the check for the builder as well here since we need to have something to call `toQuery()` on. Can delete that check and rely on our validation method (as soon as we are going to use it)